File: app/src/test/java/info/nightscout/androidaps/utils/StringUtilsTest.java
package info.nightscout.androidaps.utils;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class StringUtilsTest {

    @Test
    public void removeSurroundingQuotesTest() {
        String compareString = "test";

        assertEquals(compareString, StringUtils.removeSurroundingQuotes(compareString));
        assertEquals(compareString, StringUtils.removeSurroundingQuotes("\"" + compareString + "\""));
        assertEquals("\"" + compareString, StringUtils.removeSurroundingQuotes("\"" + compareString));

        compareString = "te\"st";
        assertEquals(compareString, StringUtils.removeSurroundingQuotes(compareString));
        assertEquals(compareString, StringUtils.removeSurroundingQuotes("\"" + compareString + "\""));
        assertEquals("\"" + compareString, StringUtils.removeSurroundingQuotes("\"" + compareString));
    }
}

File: core/src/test/java/info/nightscout/androidaps/utils/StringUtilsTest.kt
package info.nightscout.androidaps.utils

import org.junit.Assert
import org.junit.Test

class StringUtilsTest {

    @Test fun removeSurroundingQuotesTest() {
        var compareString = "test"
        Assert.assertEquals(compareString, StringUtils.removeSurroundingQuotes(compareString))
        Assert.assertEquals(compareString, StringUtils.removeSurroundingQuotes("\"" + compareString + "\""))
        Assert.assertEquals("\"" + compareString, StringUtils.removeSurroundingQuotes("\"" + compareString))
        compareString = "te\"st"
        Assert.assertEquals(compareString, StringUtils.removeSurroundingQuotes(compareString))
        Assert.assertEquals(compareString, StringUtils.removeSurroundingQuotes("\"" + compareString + "\""))
        Assert.assertEquals("\"" + compareString, StringUtils.removeSurroundingQuotes("\"" + compareString))
    }
}